corehq/apps/hqmedia/tasks.py

```diff
@@ -158,6 +176,23 @@ def build_application_zip(include_multimedia_files, include_index_files, app,
                     z.writestr(path, data, file_compression)
                     progress += file_progress / file_count
                     DownloadBase.set_progress(build_application_zip, progress, 100)
+
+        # Integrity check that all media files present in media_suite.xml were added to the zip
+        if toggles.CAUTIOUS_MULTIMEDIA.enabled(app.domain) or app.domain in ['icds', 'icds-cas']:
+            with open(fpath, 'rb') as tmp:
+                with zipfile.ZipFile(tmp, "r") as z:
+                    media_suites = [f for f in z.namelist() if re.search(r'\bmedia_suite.xml\b', f)]
+                    if len(media_suites) == 1:
+                        with z.open(media_suites[0]) as media_suite:
+                            from corehq.apps.app_manager.xform import parse_xml
+                            parsed = parse_xml(media_suite.read())
+                            resources = {node.text for node in
+                                         parsed.findall("media/resource/location[@authority='local']")}
+                            names = z.namelist()
+                            missing = [r for r in resources if re.sub(r'^\.\/', '', r) not in names]
+                            for m in missing:
+                                errors.append(_('Media file missing from CCZ: {}').format(m))
```

Comment:
One thing we would need to do here would be to delete the zip file created as well?
Since the zip file is present it would let the user download the file in the next attempt without error message as seen [here](https://dimagi-dev.atlassian.net/browse/ICDS-291?focusedCommentId=14628&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-14628)